sdk/examples/xo_go/src/sawtooth_xo/handler/handler.go

```diff
@@ -0,0 +1,322 @@
+	game, err := loadGame(name, state)
+	if err != nil {
+		return err
+	}
+	if game != nil {
+		return &processor.InvalidTransactionError{"Game already exists"}
+	}
+
+	game = &Game{
+		Board:   "---------",
+		State:   "P1-NEXT",
+		Player1: "",
+		Player2: "",
+		Name:    name,
+	}
+
+	return saveGame(game, state)
+}
+
+func applyTake(name string, space int, player string, state *processor.State) error {
+	game, err := loadGame(name, state)
+	if err != nil {
+		return err
+	}
+	if game == nil {
+		return &processor.InvalidTransactionError{"Take requires an existing game"}
+	}
+	if game.State == "P1-WIN" || game.State == "P2-WIN" || game.State == "TIE" {
+		return &processor.InvalidTransactionError{"Game has ended"}
+	}
+
+	// Assign players if new game
+	if game.Player1 == "" {
+		game.Player1 = player
+	} else if game.Player2 == "" {
+		game.Player2 = player
+	}
+
+	if game.Board[space-1] != '-' {
+		return &processor.InvalidTransactionError{"Space already taken"}
+	}
+
+	if game.State == "P1-NEXT" && player == game.Player1 {
+		boardRunes := []rune(game.Board)
+		boardRunes[space-1] = 'X'
+		game.Board = string(boardRunes)
+		game.State = "P2-NEXT"
+	} else if game.State == "P2-NEXT" && player == game.Player2 {
+		boardRunes := []rune(game.Board)
+		boardRunes[space-1] = 'O'
+		game.Board = string(boardRunes)
+		game.State = "P1-NEXT"
+	} else {
+		return &processor.InvalidTransactionError{
+			fmt.Sprintf("Not this player's turn: '%v'", player)}
+	}
+
+	if isWin(game.Board, 'X') {
+		game.State = "P1-WIN"
+	} else if isWin(game.Board, 'O') {
+		game.State = "P2-WIN"
+	} else if !strings.Contains(game.Board, "-") {
+		game.State = "TIE"
+	}
+
+	return saveGame(game, state)
+}
+
+func isWin(board string, letter byte) bool {
+	wins := [8][3]int{
+		{1, 2, 3}, {4, 5, 6}, {7, 8, 9},
+		{1, 4, 7}, {2, 5, 8}, {3, 6, 9},
+		{1, 5, 9}, {3, 5, 7},
+	}
+
+	for _, win := range wins {
+		if board[win[0]-1] == letter && board[win[1]-1] == letter && board[win[2]-1] == letter {
+			return true
+		}
+	}
+	return false
+}
+
+func unpackPayload(payloadData []byte) (*XoPayload, error) {
+	if payloadData == nil {
+		return nil, &processor.InvalidTransactionError{"Must contain payload"}
+	}
+
+	parts := strings.Split(string(payloadData), ",")
+	if len(parts) != 3 {
+		return nil, &processor.InvalidTransactionError{"Payload is malformed"}
+	}
+
+	payload := XoPayload{}
+	payload.Name = parts[0]
+	payload.Action = parts[1]
+
+	if len(payload.Name) < 1 {
+		return nil, &processor.InvalidTransactionError{"Name is required"}
+	}
+
+	if len(payload.Action) < 1 {
+		return nil, &processor.InvalidTransactionError{"Action is required"}
+	}
+
+	if payload.Action == "take" {
+		space, err := strconv.Atoi(parts[2])
+		if err != nil {
+			return nil, &processor.InvalidTransactionError{
+				fmt.Sprintf("Invalid Space: '%v'", parts[2])}
+		}
+		payload.Space = space
+	}
+
+	if strings.Index(payload.Name, "|") != -1 {
```

Comment:
You used strings.Contains above but not here. Not sure why.
`strings.Contains(payload.Name, "|")`